### DISCUSSION TITLE
How to upgrade to latest version?

### DISCUSSION THREAD

**Discussion by stacy-c** [janoside/btc-rpc-explorer#335] [Q&A]
Totally newb question but I don't know how to upgrade BTC-RPC-Explorer to the latest version. I am running version **16e387be89 (2019-02-09)**.  Suggestions please?

**Comment by janoside** [comment]
It depends...how did you install it in the first place?

If you installed via `npm`, i.e. by using the command `npm install -g btc-rpc-explorer`, then you can use a command of the following form to install a specific version (in the example, version 3.1.1):

    npm install -g btc-rpc-explorer@3.1.1

If, instead, you installed via source, by using a `git clone` command, then you can update by going to the sourcecode directory (the target of your original `git clone`) and running the below commands

    git pull
    npm install
    npm start

Lastly, if you installed via a third-party "package", such as a myNode, nodl, or umbrel, you will likely need to wait until the maintainer of your particular project makes an update available (though I don't know the details of all the projects that re-package this tool).

Does one of the above apply for you, or did you use a different installation method?

**Reply by stacy-c** [reply]
This is it. I assume that running the command below will update the project but retain all of my configurations? Thank you for your response.

`npm install -g btc-rpc-explorer@3.1.1`

**Reply by janoside** [reply]
Correct - your configurations should be unaffected by this command.

**Reply by stacy-c** [reply]
Excellent ... again, thank you for your help.

**Reply by stacy-c** [reply]
I ran the command, did `npm start` and refreshed the page but it is still running the old version. Any suggestions? Thanks.

**Reply by janoside** [reply]
Hmmm, not sure. After running the `npm install` command did the output indicate success? Additionally, when you start up the app via `npm start` what output do you see in the log?

**Reply by stacy-c** [reply]
I don't know how to make sense of the log

**Reply by stacy-c** [reply]
I had the explorer running for quite a while and I like it a lot, but you don't have to bother with this if you don't want to. I was just hoping to update it and thought it would be a smooth process.
